Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp

```diff
@@ -269,3 +298,149 @@ void URRFloatingMovementComponent::StopMovementImmediately()
     AngularVelocity = FVector::ZeroVector;
     Super::StopMovementImmediately();
 }
+
+void URRFloatingMovementComponent::SetRVOAvoidanceEnabled(bool bEnabled)
+{
+    if (bUseRVOAvoidance != bEnabled)
+    {
+        bUseRVOAvoidance = bEnabled;
+        // Reset id, RegisterMovementComponent call is required to initialize update timers in avoidance manager
+        RVOAvoidanceUID = 0;
+
+        if (bEnabled && GetOwner())
+        {
+            RVOAvoidanceManager->RegisterMovementComponent(this, RVOAvoidanceWeight);
+        }
+    }
+}
+
+void URRFloatingMovementComponent::SetUpdatedComponent(USceneComponent* InNewUpdatedComponent)
+{
+    Super::SetUpdatedComponent(InNewUpdatedComponent);
+    if (bUseRVOAvoidance)
+    {
+        RVOAvoidanceManager->RegisterMovementComponent(this, RVOAvoidanceWeight);
+    }
+}
+
+void URRFloatingMovementComponent::UpdateDefaultAvoidance()
+{
+    if (!bUseRVOAvoidance)
+    {
+        return;
+    }
+
+    if (RVOAvoidanceManager && !bRVOAvoidanceRecentlyUpdated)
+    {
+        RVOAvoidanceManager->UpdateRVO(this);
+
+        //Consider this a clean move because we didn't even try to avoid.
+        SetRVOAvoidanceVelocityLock(RVOAvoidanceManager->LockTimeAfterClean);
+    }
+
+    //Reset for next frame
+    bRVOAvoidanceRecentlyUpdated = false;
+}
+
+void URRFloatingMovementComponent::SetRVOAvoidanceVelocityLock(float InDuration)
+{
+    if (RVOAvoidanceManager)
+    {
+        RVOAvoidanceManager->OverrideToMaxWeight(RVOAvoidanceUID, InDuration);
+    }
+    RVOAvoidanceLockVelocity = Velocity;
+    RVOAvoidanceLockTimeout = InDuration;
+}
+
+void URRFloatingMovementComponent::CalculateRVOAvoidanceVelocity(float InDeltaTime)
+{
+    if (RVOAvoidanceWeight >= 1.f || !RVOAvoidanceManager || !GetOwner())
+    {
+        return;
+    }
+
+    if (GetOwner()->GetLocalRole() != ROLE_Authority)
+    {
+        return;
+    }
+
+#if RAPYUTA_SIM_DEBUG
```

Comment:
const bool bRVODebugEnabled = RAPYUTA_SIM_DEBUG && RVOAvoidanceManager->IsDebugEnabled(RVOAvoidanceUID);
then use only bRVODebugEnabled for checks below

Reply:
OK